brainio/assemblies.py

```diff
@@ -321,4 +359,83 @@ def get_levels(assembly):
         if isinstance(value, IndexVariable) and value.level_names:
             for level in value.level_names:
                 levels.append(level)
-    return levels
\ No newline at end of file
+    return levels
+
+
+class AssemblyLoader:
+    """
+    Loads an assembly from a file.
+    """
+
+    def __init__(self, cls, file_path, group=None, **kwargs):
+        self.assembly_class = cls
+        self.file_path = file_path
+        self.group = group
+
+    def load(self):
+        if self.group is None:
+            data_array = xr.open_dataarray(self.file_path)
+        else:
+            data_array = xr.open_dataarray(self.file_path, group=self.group)
+        result = data_array
+        result = self.assembly_class(data=result)
+        return result
+
+
+class StimulusMergeLoader(AssemblyLoader):
+    """
+    Loads an assembly and add metadata from a stimulus set.
```

Comment:
from the class name I thought this would load a stimulus set -- maybe something like `AssemblyLoaderMergeStimuli`?

Reply:
Done.